CMakeLists.txt

```diff
@@ -254,131 +272,77 @@ endif()
 #   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
 #   BOOST_ROOT - root of Boost installation
 #   BOOST_INCLUDEDIR - include directory for Boost
+#   CEREAL_INCLUDE_DIR - include directory for cereal
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+if (CMAKE_CROSSCOMPILING)
+  get_deps(https://github.com/xianyi/OpenBLAS/releases/download/v0.3.13/OpenBLAS-0.3.13.tar.gz OpenBLAS OpenBLAS-0.3.13.tar.gz)
+  if (NOT MSVC)
```

Comment:
I noticed that this runs every time configuration happens, which is probably not what we want.  If you want to see, just try typing `cmake ../` a second time after configuring CMake.  Maybe a solution here is to have `get_deps()` first check to see if the target directory exists, and if so, it does nothing but sets some variable like `DOWNLOAD_ALREADY_EXISTS`.  Then, here, we enclose the `execute_process()` in a `if (NOT DOWNLOAD_ALREADY_EXISTS)`.  That's just one idea; I can think of a few ways to do it.  Let me know what you think. :+1:

Reply:
I will add two check, one for the package and the other for the library.
In this case, if the library is already downloaded, then it will check if it is already compiled.
We do not want to run into a situation when the library is only downloaded but not compiled.

Reply:
@shrit good call; I would be okay with just assuming that if it wasn't downloaded it needed to be downloaded and compiled, but two checks is fine (and safer) too. :+1: